Assets/Scripts/Game/MagicAndEffects/Effects/Mysticism/Teleport.cs

```diff
@@ -164,11 +158,10 @@ void TeleportPlayer()
                 // Need to load some other part of the world again - player could be anywhere
                 PlayerEnterExit.OnRespawnerComplete += PlayerEnterExit_OnRespawnerComplete;
                 playerEnterExit.RestorePositionHelper(anchorPosition, false);
-            }
 
-            // When moving anywhere other than same interior trigger a fade so transition appears smoother
-            if (!sameInterior)
```

Comment:
That's good. And since we're not checking this bool twice, can probably do away with:

    bool sameInterior = IsSameInterior();
    if (sameInterior)
    ...

and just change if statement to:

    if (IsSameInterior())
    ...

It cleans up another line and removed a value checked only once now.